Source/KSCrash/Source/KSCrash/Recording/Tools/BSG_RFC3339DateTool.m

```diff
@@ -26,29 +26,39 @@
 
 @implementation BSG_RFC3339DateTool
 
-static NSDateFormatter *bsg_g_formatter;
+static NSString *const kDateFormatterKey = @"RfcDateFormatter";
 
-+ (void)initialize {
-    bsg_g_formatter = [[NSDateFormatter alloc] init];
-    NSLocale *locale =
-        [[NSLocale alloc] initWithLocaleIdentifier:@"en_US_POSIX"];
-    [bsg_g_formatter setLocale:locale];
-    [bsg_g_formatter setDateFormat:@"yyyy'-'MM'-'dd'T'HH':'mm':'ss'Z'"];
-    [bsg_g_formatter setTimeZone:[NSTimeZone timeZoneForSecondsFromGMT:0]];
++ (NSDateFormatter *)sharedInstance {
+    @synchronized(self) {
```

Comment:
Doesn't the thread dict negate the need to synchronize access? `self` here is the class rather than an instance.

Reply:
@kattrali that's true - I've updated the PR.